### DISCUSSION TITLE
Color Reduction to 256 colors out of a fixed 512 color palette

### DISCUSSION THREAD

**Discussion by HeartWareDK** [ImageMagick/ImageMagick#2048] [General]
Is it possible to process an image (PNG/BMP 24bit RGB888 format) into a new 8bit palette format PNG/BMP with 256 colors selected from a given 512-entries palette (RGB333), and using dithering to "simulate" the remaining colors?

**Comment by snibgo** [comment]
You can "-remap" an image to use only colours that are present in another image. That other image can contain 256 colours or 512 or any number you want.

IM has no simple operation to select the best 256 colours from a 512-colour image. But you could remap to the 512 colours, then find the 256 most used colours, make a map of those, restore the other colours to the original input, and remap to the 256 map.

This would need a script.
